### PR TITLE
Element: Add gateway

### PR DESCRIPTION
**Pull request by davidsantoso**
@duff and/or @rwdaigle: Adding the Element gateway. Pretty standard, nothing really to point out.

### CODE REVIEW

File: lib/active_merchant/billing/gateways/element.rb [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+        response["expressresponsemessage"]
+      end
+
+      def authorization_from(response)
+        response["transaction"]["transactionid"]
+      end
+
+      def error_code_from(response)
+        unless success_from(response)
+          # TODO: lookup error code for this response
```
Comment (rwdaigle) [inline lib/active_merchant/billing/gateways/element.rb]: Should this be filled in?

Comment (davidsantoso) [inline lib/active_merchant/billing/gateways/element.rb]: Ahh, good catch. Actually not supposed to be filled in. I'm just going to :hocho: it.

File: lib/active_merchant/billing/gateways/element.rb [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+        end
+      end
+
+      def supports_scrubbing?
+        true
+      end
+
+      def scrub(transcript)
+        transcript.
+          gsub(%r((<AccountID>).+?(</AccountID>))i, '\1[FILTERED]\2').
```
Comment (davidsantoso) [inline lib/active_merchant/billing/gateways/element.rb]: I don't think it is come to think of it. Updating...

**Comment by davidsantoso** [comment]
@rwdaigle it does. Should `store` be done by default in addition to the other actions if it's available?

File: lib/active_merchant/billing/gateways/element.rb (hunk 2 above)
Comment (girasquid) [inline lib/active_merchant/billing/gateways/element.rb]: :+1: for not scrubbing if it's not actually PCI-level sensitive - accountID can be really handy for debugging if you need to look something up in logs and send to support staff somewhere

Comment (davidsantoso) [inline lib/active_merchant/billing/gateways/element.rb]: :+1:

**Comment by rwdaigle** [comment]
> Should store be done by default in addition to the other actions if it's available?

I believe that is Spreedly's current approach, yes (let me know if we need to update our internal docs to reflect this if it's unclear)?

**Comment by davidsantoso** [comment]
@rwdaigle updated with `store` and added the ability to charge via a an Element `PaymentAccountID`. Let me know if you have any other feedback!

File: lib/active_merchant/billing/gateways/element.rb [hunk 3]
```diff
@@ -0,0 +1,327 @@
...
+
+      def supports_scrubbing?
+        true
+      end
+
+      def scrub(transcript)
+        transcript.
+          gsub(%r((<AccountToken>).+?(</AccountToken>))i, '\1[FILTERED]\2').
+          gsub(%r((<CardNumber>).+?(</CardNumber>))i, '\1[FILTERED]\2').
+          gsub(%r((<CVV>).+?(</CVV>))i, '\1[FILTERED]\2')
```
Comment (rwdaigle) [inline lib/active_merchant/billing/gateways/element.rb]: I believe we need `RoutingNumber` and `AccountNumber` here as well?

**Comment by davidsantoso** [comment]
@rwdaigle wow good catch! Definitely need to scrub those. I just updated again!

**Comment by rwdaigle** [comment]
> wow good catch!

10/10 would PR again?

**Comment by rwdaigle** [comment]
Closed w/ https://github.com/activemerchant/active_merchant/commit/7519bc780e704e8e801a5483cb4a235d6b9e55f5

**Pull request closed without merging**